source/slang/slang-check-expr.cpp

```diff
@@ -1529,6 +1538,51 @@ namespace Slang
         return CheckInvokeExprWithCheckedOperands(expr);
     }
 
+    Expr* SemanticsExprVisitor::visitTryExpr(TryExpr* expr)
+    {
+        auto prevTryClauseType = expr->tryClauseType;
+        m_enclosingTryClauseType = expr->tryClauseType;
+        expr->base = CheckTerm(expr->base);
+        m_enclosingTryClauseType = prevTryClauseType;
+        expr->type = expr->base->type;
+        if (as<ErrorType>(expr->type))
+            return expr;
+        
+        auto parentFunc = this->m_parentFunc;
+        // TODO: check if the try clause is catched.
+        // For now we assume all `try`s are not catched (because we don't have catch yet).
+        if (!parentFunc)
+        {
+            getSink()->diagnose(expr, Diagnostics::uncachedTryCallInNonThrowFunc);
+            return expr;
+        }
+        if (parentFunc->errorType->equals(m_astBuilder->getVoidType()))
```

Comment:
Semantically, we probably shouldn't use the `void` type to represent the "doesn't throw" case, but for now this is probably fine.

In practice `void` is our "unit" type, and thus has a single representable value. It could thus be reasonable for a function to declare that it can `throw` an error of type `void` (in which case there is only one possible error value it can throw).

Eventually we will need/want to have a "bottom" type (probably called `Never`), and *that* would be the appropriate error type to use for a function that cannot `throw`.